src/main/java/net/openhft/chronicle/wire/WireMarshaller.java

```diff
@@ -1796,6 +1910,22 @@ protected void setValue(Object o, ValueIn read, boolean overwrite) {
             throw new UnsupportedOperationException();
         }
 
+        @Override
+        protected void setDefaultValue(Object ignored, Object o) {
+            // TODO very limited form of deep reset, check for immutability
+            if (defaultValue != null && defaultValue.isEmpty()) {
+                Map<?, ?> map = unsafeGetObject(o, offset);
+                if (map == null) {
+                    map = collectionSupplier.get();
+                    unsafePutObject(o, offset, map);
+                }
+                map.clear();
+                return;
+            }
+
+            unsafePutObject(o, offset, defaultValue);
```

Comment:
Again this seems fairly dangerous. Interested to know how it compares to existing behaviour